sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala

```diff
@@ -178,5 +180,9 @@ class ExpressionSet protected(
        |baseSet: ${baseSet.mkString(", ")}
        |originals: ${originals.mkString(", ")}
      """.stripMargin
+
+  /** Returns a length limited string that must be used for logging only. */
+  def simpleString(maxFields: Int): String =
+    s"Set(${truncatedString(originals.map(_.simpleString(maxFields)).toSeq, ", ", maxFields)})"
```

Comment:
I guess you convert elements via `simpleString` even if it could be not needed. Let's imagine `originals` has 10000 elements, and `maxFields` is 5. Any any case you converts all 10000 elements to strings.

How about to add override `truncatedString` w/ additional parameter: a function which converts `T` to `String`:

```scala
truncatedString(originals.toSeq, "Set(", ", ", ")", maxFields, _.simpleString(maxFields))
```

Reply:
Had the exact same in mind, will pick that up. Default value is toString